packages/app/src/navigation/SidebarNavigation.vue

```diff
@@ -138,16 +152,54 @@ mutation SideBarNavigation_SetPreferences ($value: String!) {
   }
 }`
 
-gql`
-query SideBarNavigation {
-  ...SidebarNavigationHeader
-  ...SidebarNavigation
-}
-`
+const props = defineProps<{ gql: SidebarNavigationFragment | undefined, isLoading: boolean }>()
 
 const NAV_EXPAND_MIN_SCREEN_WIDTH = 1024
 
-const query = useQuery({ query: SideBarNavigationDocument })
+const loginConnectStore = useLoginConnectStore()
+
+const debugBadge = computed<Badge | undefined>(() => {
+  if (props.isLoading) {
+    return undefined
+  }
+
+  const showNewBadge = isAllowedFeature('debugNewBadge', loginConnectStore)
+
+  if (props.gql?.currentProject?.cloudProject?.__typename === 'CloudProject') {
+    const { status, testsForReview } = props.gql.currentProject.cloudProject.runByNumber || {}
+
+    if (status === 'NOTESTS' || status === 'RUNNING') {
+      return showNewBadge ? { value: t('sidebar.debug.new'), status: 'success', label: t('sidebar.debug.debugFeature') } : undefined
```

Comment:
The object for the "new" badge is duplicated on line 192.  Pull this out into a variable?

Reply:
Addressed in [4c21f5f](https://github.com/cypress-io/cypress/pull/25152/commits/4c21f5f4d005a19a9dbe52ffac3101f9bfeb513f)